src/Simulator/IncrementalRedstoneSimulator/RedstoneComparatorHandler.h

```diff
@@ -54,7 +54,14 @@ class cRedstoneComparatorHandler : public cRedstoneHandler
 
 			virtual bool Item(cBlockEntity * a_BlockEntity) override
 			{
-				auto & Contents = static_cast<cBlockEntityWithItems *>(a_BlockEntity)->GetContents();
+				// Skip BlockEntities that don't have slots
+				auto BlockEntityWithItems = dynamic_cast<cBlockEntityWithItems *>(a_BlockEntity);
+				if (BlockEntityWithItems != nullptr)
```

Comment:
You are returning `false` if this is **not** a `nullptr` ?? Is this right? Before, you were checking for falsyness.